sessiontoken_test.go

```diff
@@ -1,19 +1,55 @@
 package signalfx
 
 import (
+	"fmt"
 	"net/http"
+	"net/http/httptest"
+	"net/url"
 	"testing"
 
 	"github.com/signalfx/signalfx-go/sessiontoken"
 
 	"github.com/stretchr/testify/assert"
 )
 
+func verifyNoTokenRequest(t *testing.T, method string, status int, params url.Values, resultPath string) func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Rather than duplicate `verifyRequest` could we add a `bool` argument as to the presence or absence of the token? I believe it could be:
```
func verifyRequest(t *testing.T, method string, expectToken bool, status int, params url.Values, resultPath string) func(w http.ResponseWriter, r *http.Request) {
	return func(w http.ResponseWriter, r *http.Request) {
		if val, ok := r.Header[AuthHeaderKey]; ok {
			assert.Equal(t, []string{TestToken}, val, "Incorrect auth token in headers")
		} else {
  			if expectToken {
  			  assert.Fail(t, "Failed to find auth token in headers")
  			}
		}```

I understand it'll have some impact in needing to modify every call site, but it seems less burdensome later?

Reply:
I had that idea initially but for the reasons you stated, I opted for this alternative. You’re right though, in the long wrong this is a cleaner solution. I’ll update my PR

Reply:
lol @ long wrong typo

Excellent, thanks for the attention here!

Reply:
Haha that's what happens when you respond to PRs on your phone waiting for your car to be washed :P

Reply:
Alrighty, that's now ready for review. Only 20 file changes!